src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -302,7 +302,8 @@
 ;; Functions to ensure that the catalog structure is coherent.
 
 (def ^:const tag-pattern
-  #"\A[a-z0-9_][a-z0-9_:\-.]*\Z")
+  ; https://puppet.com/docs/puppet/6.4/lang_reserved.html#tags
+  #"\A[[\p{L}&&[^\p{Lu}]]\p{N}_][[\p{L}&&[^\p{Lu}]]\p{N}_:\-.]*\Z")
```

Comment:
Wonder if it would be helpful or harmful to add something like this to the comment:

```clojure
;; This regex is intended to match this ruby pattern which was specified
;; there: \A[[:alnum:]_][[:alnum:]_:.-]*\Z
```

So that we'll know what was originally intended, even if the puppet standard changes, or maybe that'd be better in the commit message, if anywhere.  Not sure.

Reply:
It may be harmful in the comment, if there's a change on the Ruby side, hopefully someone will follow the link to find an updated regex as opposed to checking the regex in the comment. But adding it in the commit message seems like a fine way to record the actual regex we are matching.